pkg-r/R/QueryChat.R

```diff
@@ -668,13 +668,22 @@ querychat_app <- function(
   categorical_threshold = 20,
   extra_instructions = NULL,
   prompt_template = NULL,
-  cleanup = TRUE,
+  cleanup = NA,
   bookmark_store = "url"
 ) {
   if (is_missing(table_name) && is.data.frame(data_source)) {
     table_name <- deparse1(substitute(data_source))
   }
 
+  if (is.na(cleanup) && interactive() && !in_shiny_session()) {
+    if (
+      is.data.frame(data_source) ||
+        inherits(data_source, "DataFrameSource")
+    ) {
+      cleanup <- TRUE
+    }
+  }
```

Comment:
Should we do the analogous thing for Python?

Reply:
Yeah